avocado_vt/plugins/vt_runner.py

```diff
@@ -1,53 +1,54 @@
-import logging
 import multiprocessing
 import os
-import sys
-import tempfile
 import time
 import traceback
 
-from avocado.core import exceptions, nrunner, output, teststatus
+from avocado.core import exceptions, nrunner, teststatus
 from avocado.core.runners.utils import messages
-from avocado.utils import astring, path, stacktrace
+from avocado.core.test_id import TestID
+from avocado.utils import astring
 
-from avocado_vt import utils
-from virttest import (data_dir, env_process, error_event, utils_env, utils_misc,
-                      utils_params, version, funcatexit)
+from avocado_vt import test
 
 
-class VirtTest(utils.TestUtils):
+class VirtTest(test.VirtTest):
 
     def __init__(self, queue, runnable):
-        self.__vt_params = utils_params.Params(runnable.kwargs)
         self.queue = queue
-        self.config = runnable.config
-        self.tmpdir = tempfile.mkdtemp()
-        self.logdir = os.path.join(self.tmpdir, 'results')
-        path.init_dir(self.logdir)
-        self.logfile = os.path.join(self.logdir, 'debug.log')
-        self.log = output.LOG_JOB
-        self.log_level = runnable.config.get('job.output.loglevel',
-                                             logging.DEBUG)
-        self.env_version = utils_env.get_env_version()
-        self.iteration = 0
-        self.background_errors = error_event.error_events_bus
-        # clear existing error events
-        self.background_errors.clear()
-        self.debugdir = self.logdir
-        self.bindir = data_dir.get_root_dir()
-        self.virtdir = os.path.join(self.bindir, 'shared')
-
-    @property
-    def params(self):
-        return self.__vt_params
-
-    def run_test(self):
-        status = "ERROR"
+        vt_params = runnable.kwargs
+        vt_params['job_env_cleanup'] = 'no'
+        kwargs = {'name': TestID(1, runnable.uri),
+                  'config': runnable.config,
+                  'vt_params': vt_params}
+        super().__init__(**kwargs)
+
+    def _save_log_dir(self):
```

Comment:
This looks similar to a proposal of mine about the test's `datadir` in Avocado, which was initially rejected because it was assumed that `avocado jobs get-output-files` would drive that instead.

IMO there are valid rationales for supporting both approaches:

1. automatic push (like proposed here)
2. manual pull (like with `avocado jobs get-output-files`)

One major downside of the automatic push approach has to do with large results created by tests, **and** with the fact that we are using (and only using) the messages as transports.  For the majority of executions we have:

a. an output dir being created by the task (`11:52:09 DEBUG| Task "1-io-github-autotest-qemu.boot" started message: "{'status': 'started', 'time': 443943.220665491, 'output_dir': '/var/tmp/.avocado-task-pusf_jua', 'id': '1-io-github-autotest-qemu.boot'}"`)
b. no results available to the user under the job directory **until** the test finishes
c. the serialization of all files in the output dir
d. the copy of all files from the task's `outputdir` to the test results dir under the job structure

This is very wasteful, given that it's the most common use case, we IMO we need to address it.  And it's relevant not only for the "Process" spawner, but even with a Podman spawner, a volume (`-v`) could be used to optimize steps a-d.

And just to make it clear, those points are not necessarily blockers to this PR.

Reply:
Indeed, I mentioned something similar in https://github.com/avocado-framework/avocado-vt/pull/3214#issuecomment-919833151. IIUC, lengthy unattended installation or GUI (e.g. operating on Windows) tests produce lots of files used either for checking how the test is doing or for later debugging and while the desire to handle various remote hosts is reasonable my concern is that a test runner would want to at least know some of the progress at the frontend which in this case is the results directory.

Reply:
I know that this is not the perfect solution, but I wanted to make those files available for users, because for now they are hidden in `/tmp`, and then we can discuss how to make it better.

Reply:
I agree, this makes complete sense and the current lack of the files is essentially a regression that has to be fixed first.